pkg/resourceinterpreter/default/native/retain.go

```diff
@@ -122,3 +126,23 @@ func retainJobSelectorFields(desired, observed *unstructured.Unstructured) (*uns
 	}
 	return desired, nil
 }
+
+func retainDeploymentFields(desired, observed *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	labels, exist, err := unstructured.NestedStringMap(desired.Object, "metadata", "labels")
```

Comment:
I think this is a general logic, not just for `Deployment`, so I prefer to set it as a default function, similar to `reflectWholeStatus`.

Reply:
So we don't need to write the same logic again for `pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CloneSet`.

Reply:
Operations on the replicas of Kubernetes Workload resources can be universal. Therefore, operations on replicas can be extracted as an independent function, such as `retainK8sWorkloadReplicas`.

Reply:
> So we don't need to write the same logic again for `pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/CloneSet`.

Are you currently using HPA to control CloneSets?

Reply:
I haven't used it in prod. 

However, I know that in the `OpenKruise` community, there are already users who use HPA to control CloneSet.

Reply:
If needed, we can submit a single pr to set the Retain operation for the `CloneSet` and other resources.